internal/app/backend/backend_service.go

```diff
@@ -204,18 +204,32 @@ func matchesFromHTTPMMF(ctx context.Context, profile *pb.MatchProfile, client *h
 		}
 	}()
 
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "failed to read from response body for profile %s: %s", profile.Name, err.Error())
-	}
+	dec := json.NewDecoder(resp.Body)
+	proposals := make([]*pb.Match, 0)
+	for {
+		var item struct {
+			Result json.RawMessage        `json:"result"`
+			Error  map[string]interface{} `json:"error"`
+		}
 
-	pbResp := &pb.RunResponse{}
-	err = json.Unmarshal(body, pbResp)
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "failed to unmarshal response body to response pb for profile %s: %s", profile.Name, err.Error())
+		err := dec.Decode(&item)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, status.Errorf(codes.FailedPrecondition, "failed to decode RunResponse from JSON: %s", err.Error())
```

Comment:
FailedPrecondition doesn't seem like the correct error code here.  A failure here isn't likely to be a problem with system state which needs to be corrected. (eg, if this error is a timeout, that's not at issue here.)
I think these should all be Unavailable?
See https://github.com/grpc/grpc-go/blob/master/codes/codes.go

Reply:
I agree that this one should be Unavailable as it reflects the service is currently unavailable. For the second one, I changed it to directly return the error since it is from a service that we own. 

I think `FailedPrecondition` applies to the third case since the error most likely means the client wrote a bad mmf and should not retry until the system state has been explicitly fixed